crates/ursa-rpc-service/src/api.rs

```diff
@@ -97,51 +108,43 @@ where
     pub store: Arc<UrsaStore<S>>,
     pub network_send: Sender<NetworkCommand>,
     pub provider_send: Sender<ProviderCommand>,
+    pending_requests: PendingRequests,
+    client: Arc<Client>,
+    origin_config: OriginConfig,
 }
 
 #[async_trait]
 impl<S> NetworkInterface for NodeNetworkInterface<S>
 where
     S: Blockstore + Store + Send + Sync + 'static,
 {
-    async fn get(&self, cid: Cid) -> Result<Option<Vec<u8>>> {
-        if !self.store.blockstore().has(&cid)? {
-            info!("Requesting block with the cid {cid:?}");
-            let (sender, receiver) = oneshot::channel();
-            let request = NetworkCommand::GetBitswap { cid, sender };
-
-            // use network sender to send command
-            self.network_send.send(request)?;
-            if let Err(e) = receiver.await? {
-                return Err(anyhow!(
-                    "The bitswap failed, please check server logs {:?}",
-                    e
-                ));
+    async fn get(&self, cid: Cid) -> Result<Vec<u8>> {
+        let new = self.check_blockstore(cid).await?;
+        let content =
+            self.store.blockstore().get(&cid)?.ok_or_else(|| {
+                anyhow!("content was fetched but could not be found in blockstore")
+            })?;
+        if new {
+            let size = content.len() as u64;
```

Comment:
Doesn't `self.store.blockstore().get` return just one block? if so `content.len()` would be the length of just one block instead of the length/size of the file?

Reply:
That's correct. I'm not sure how much of a use this method is though, as it's not currently used anywhere in the codebase. See https://github.com/fleek-network/ursa/pull/278#discussion_r1071471095 for my response on fixing the size calculation

Reply:
addressed here: https://github.com/fleek-network/ursa/pull/278/commits/ebb750854c81e7ec038c6e740b0cc8db0d4a3085